File: benchmark/comparison/results/haiku_ts/03.ts
type Color = "Red" | "Green" | "Blue" | "Yellow";

function toHex(color: Color): string {
  switch (color) {
    case "Red": return "#FF0000";
    case "Green": return "#00FF00";
    case "Blue": return "#0000FF";
    case "Yellow": return "#FFFF00";
  }
}

function isPrimary(color: Color): boolean {
  return color === "Red" || color === "Green" || color === "Blue";
}

// Test cases
const colors: Color[] = ["Red", "Green", "Blue", "Yellow"];

for (const color of colors) {
  console.log(toHex(color));
}

for (const color of colors) {
  console.log(isPrimary(color) ? "true" : "false");
}

File: benchmark/comparison/results/adv_py/03.py
class Color:
    Red = "Red"
    Green = "Green"
    Blue = "Blue"
    Yellow = "Yellow"

def to_hex(c):
    mapping = {
        Color.Red: "#FF0000",
        Color.Green: "#00FF00",
        Color.Blue: "#0000FF",
        Color.Yellow: "#FFFF00",
    }
    return mapping[c]

def is_primary(c):
    return c in (Color.Red, Color.Green, Color.Blue)

for c in [Color.Red, Color.Green, Color.Blue, Color.Yellow]:
    print(to_hex(c))

for c in [Color.Red, Color.Green, Color.Blue, Color.Yellow]:
    print("true" if is_primary(c) else "false")
